packages/shared/src/components/sidebar/MyFeedAlert.tsx

```diff
@@ -1,48 +1,106 @@
-import React, { ReactElement } from 'react';
+import React, { CSSProperties, ReactElement, ReactNode, useRef } from 'react';
+import classNames from 'classnames';
 import XIcon from '../icons/Close';
-import { Alerts } from '../../graphql/alerts';
 import { Button } from '../buttons/Button';
-import { AlertContainer, AlertCopy } from './common';
-import { Pointer, PointerColor } from '../Pointer';
+import { AlertContainer, AlertCopy, AlertWrapper } from './common';
+import Pointer, { PointerColor } from '../Pointer';
 
-const alertCopy = {
-  migrated: `Psst, your feed has a new name! We've already applied your content filters to it.`,
-  created: `🎉 Your feed is ready! Click here to manage your feed's settings`,
+interface ClassName {
+  container?: string;
+  label?: string;
+}
+
+export enum AlertPlacement {
+  Top = 'top',
+  Right = 'right',
+  Bottom = 'bottom',
+  Left = 'left',
+}
+
+const pointerClasses: Record<AlertPlacement, string> = {
+  top: 'top-0 rotate-180 -translate-y-full',
+  right: 'right-0 -rotate-90',
+  left: 'left-0 rotate-90',
+  bottom: 'bottom-0 translate-y-full',
 };
 
+const messageContainerClasses: Record<AlertPlacement, string> = {
+  top: '-translate-y-full',
+  right: 'translate-x-full',
+  left: '-translate-x-full',
+  bottom: 'translate-y-full',
+};
+
+const HORIZONTAL_OFFSET = 6;
+
 interface MyFeedAlertProps {
-  alerts: Alerts;
+  className?: ClassName;
   hideAlert: () => unknown;
+  children: ReactNode;
+  placement?: AlertPlacement;
 }
+
+const getStyle = (
+  placement: AlertPlacement,
+  { height = 0 }: Partial<DOMRect> = {},
+): CSSProperties => {
+  if (placement === AlertPlacement.Left) {
```

Comment:
I feel like we can optimize this a little bit better, maybe by using a same Record approach as above?

Reply:
Make sense, let me find a way to do it a much secure way like you suggested.